autoload/ale/linter.vim

```diff
@@ -463,6 +465,11 @@ function! ale#linter#StartLSP(buffer, linter, callback) abort
         call ale#lsp#Send(l:conn_id, ale#lsp#tsserver_message#Change(a:buffer))
     endif
 
+    let l:settings = ale#Var(a:buffer, 'lsp_settings')
+    if has_key(l:settings, a:linter['name'])
+        call ale#lsp#Send(l:conn_id, ale#lsp#message#DidChangeConfiguration(l:settings[a:linter['name']]), l:root)
+    endif
```

Comment:
If you need to run this before opening the documents, move these lines above the `OpenDocumentIfNeeded` call.

Reply:
Check `a:linter.lsp isnot# 'tsserver'` to prevent the message being sent for `tsserver`.

Reply:
I had tried moving these lines above `OpenDocumentIfNeeded`, but that didn't help. [here's](https://gist.github.com/aurieh/11d920a8094b4c5236a57c97b35c4859) the channel log.

Reply:
Beats me then. That should work.